Funcky/Extensions/EnumerableExtensions/Interleave.cs

```diff
@@ -0,0 +1,30 @@
+using System.Collections.Generic;
+using System.Diagnostics.Contracts;
+using System.Linq;
+
+namespace Funcky.Extensions
+{
+    public static partial class EnumerableExtensions
+    {
+        public static IEnumerable<T> Interleave<T>(this IEnumerable<T> sequence, params IEnumerable<T>[] otherSequences)
+            => Interleave(new[] { sequence }.Concat(otherSequences));
```

Comment:
Id rather would like to construct an immutable list here (should we include the System.Collections.Immutable Nuget?)

Reply:
Fixed with b72a4671